Source/Scene/PostProcessCollection.js

```diff
@@ -0,0 +1,596 @@
+        '../Shaders/PostProcessFilters/PassThrough',
+        '../ThirdParty/Shaders/FXAA3_11',
+        './PostProcess',
+        './PostProcessAmbientOcclusionStage',
+        './PostProcessBloomStage',
+        './PostProcessSampleMode',
+        './PostProcessTextureCache'
+    ], function(
+        Check,
+        defaultValue,
+        defined,
+        defineProperties,
+        destroyObject,
+        DeveloperError,
+        FXAAFS,
+        PassThrough,
+        FXAA3_11,
+        PostProcess,
+        PostProcessAmbientOcclusionStage,
+        PostProcessBloomStage,
+        PostProcessSampleMode,
+        PostProcessTextureCache) {
+    'use strict';
+
+    var fxaaFS =
+        '#define FXAA_QUALITY_PRESET 39 \n' +
+        FXAA3_11 + '\n' +
+        FXAAFS;
+
+    var stackScratch = [];
+
+    /**
+     * A collection of {@link PostProcess}es and/or {@link PostProcessComposite}s.
+     * <p>
+     * The texture modified by each post process is the texture rendered to by the scene or the texture rendered
+     * to by the previous post-process or composite in the collection.
+     * </p>
+     * <p>
+     * If the ambient occlusion or bloom post-processes are enabled, they will execute before all other post-processes.
+     * </p>
+     * <p>
+     * If the FXAA post-process is enabled, it will execute after all other post-processes.
+     * </p>
+     *
+     * @alias PostProcessCollection
+     * @constructor
+     */
+    function PostProcessCollection() {
+        this._processes = [];
+        this._activeProcesses = [];
+
+        this._fxaa = new PostProcess({
+            name : 'czm_FXAA',
+            fragmentShader : fxaaFS,
+            sampleMode : PostProcessSampleMode.LINEAR
+        });
+        this._ao = new PostProcessAmbientOcclusionStage();
+        this._bloom = new PostProcessBloomStage();
+
+        this._ao.enabled = false;
+        this._bloom.enabled = false;
+
+        this._processesRemoved = false;
+        this._cacheDirty = false;
+
+        this._lastLength = undefined;
+        this._aoEnabled = undefined;
+        this._bloomEnabled = undefined;
+        this._fxaaEnabled = undefined;
+
+        this._processNames = {};
+        this._textureCache = undefined;
+
+        var processNames = this._processNames;
+
+        var stack = stackScratch;
+        stack.push(this._fxaa, this._ao, this._bloom);
+        while (stack.length > 0) {
+            var process = stack.pop();
+            processNames[process.name] = process;
+            process._collection = this;
+
+            var length = process.length;
+            if (defined(length)) {
+                for (var i = 0; i < length; ++i) {
+                    stack.push(process.get(i));
+                }
+            }
+        }
+    }
+
+    defineProperties(PostProcessCollection.prototype, {
+        /**
+         * Determines if all of the post-processes in the collection are ready to be executed.
```

Comment:
Throughout the reference doc, comments, code, etc., I am more leaning towards replace "post-processes" with "post-processing stages."

Reply:
Likewise throughout where `process` is used as a variable name, I think it should be `stage`.

Reply:
Likewise, `processes` -> `stages`

And similar for others.  I think this will read a lot better.